### PR TITLE
Limit user permissions

### PR DESCRIPTION
**Pull request by SabrinaSnider**
Limited delete permissions of comments. Admins have the option to delete any individual comment and delete all comments. Users have the option to delete their own comments.

Instead of manually creating JSON string for some server responses, I switched to using a JsonObject to automatically build them (from google gson library).

I also moved user-related functions to their own javascript file for organization.

While logged in:
image

While logged out:
image

### CODE REVIEW

File: portfolio/src/main/webapp/scripts/comments.js [hunk 1]
```diff
@@ -149,15 +127,24 @@ function addComment(comment) {
     innerText: comment.message,
   })
 
+  commentItem.appendChild(commentHeader);
+  commentHeader.appendChild(commentEmail);
+  commentItem.appendChild(commentMessage);
+  
+
```
Comment (brucezheng) [inline portfolio/src/main/webapp/scripts/comments.js:134]: Nit: Remove extra newline.

File: portfolio/src/main/webapp/scripts/user.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    } else {
+      return data.email;
+    }
+  })
+}
+
+/* Helper function to check if the user is an admin */
+async function isAdmin() {
+  return fetch('/is-admin', { method: "get" }).then(response => response.json()).then(data => {
+    return data.admin;
```
Comment (brucezheng) [inline portfolio/src/main/webapp/scripts/user.js:16]: Maybe ```return data.admin === "true"``` so this actually returns a bool.

Comment (SabrinaSnider) [reply portfolio/src/main/webapp/scripts/user.js:16]: Good idea

**Comment by SabrinaSnider** [comment]
The way deleting works now is that the front-end makes a server call for permissions, then the front-end determines whether or not to display the delete buttons. I can move the permission check to the backend, but I'm not sure how I could check whether the buttons should be displayed.

**Comment by brucezheng** [comment]
Yeah, I think you need both front end and backend changes.

**Comment by brucezheng** [comment]
See this article: https://medium.com/better-programming/back-end-data-validations-73ea9004c6d7

**Comment by SabrinaSnider** [comment]
I made changed to do fewer server calls. The main ones are:

1. Seperated getComments() into two methods: getComments() which fetches comments from the server, and displayComments() which displays them on the front-end.
2. Combined the IsAdminServlet and GetUserServlet into one servlet called GetUserData. This returns both the email of the current user and the admin status. Since these values are usually used together, I figured it would be more efficient.
3. Fetched the admin status and user email prior to generating individual comments for the fron-end. Before, the fetches were made inside generating each comment, which caused extra unnecessary fetches.

**Comment by SabrinaSnider** [comment]
Last push was for deleting an old servlet (IsAdminServlet), which was replaces by GetUserDataServlet.

File: portfolio/src/main/java/com/google/sps/servlets/GetUserDataServlet.java [hunk 3]
```diff
@@ -9,21 +9,32 @@
...
     if (userService.isUserLoggedIn()) {
       String email = userService.getCurrentUser().getEmail();
-      response.getWriter().println("{\"email\": \"" + email + "\"}");
+      json.addProperty("email", email);
+
+      if (userService.isUserAdmin()) {
+        json.addProperty("admin", "true");
+      } else {
+        json.addProperty("admin", "false");
+      }
```
Comment (ccyang314) [inline portfolio/src/main/java/com/google/sps/servlets/GetUserDataServlet.java:31]: nit: this logic can be replaced by a ternary operator:

`json.addProperty("admin", userService.isUserAdmin() ? "true" : "false");`

Comment (SabrinaSnider) [reply portfolio/src/main/java/com/google/sps/servlets/GetUserDataServlet.java]: I tried doing
`userService.isUserAdmin() ? json.addProperty("admin", "true"): json.addProperty("admin", "false");`

But after looking it up, it seems like Java only allows ternary operators when the right side returns a value. (https://stackoverflow.com/questions/9450864/java-ternary-with-no-return-for-method-calling)

To clean it up, I instead condensed the if statement to one line.

Comment (SabrinaSnider) [reply portfolio/src/main/java/com/google/sps/servlets/GetUserDataServlet.java]: Oh interesting, I didn't know you could do a ternary operator in a statement!

**Pull request merged**